src/mongo/MongoScrapbook.ts

```diff
@@ -112,16 +112,13 @@ async function executeCommand(activeEditor: vscode.TextEditor, database: MongoDa
 			throw new Error(`Error near line ${err.range.start.line}, column ${err.range.start.character}: '${err.message}'. Please check syntax.`);
 		}
 
-		if (command.name === 'find') {
+		if (command.name === 'find' && !!command.collection && !command.sendToShell) {
```

Comment:
Can we move this code directly into `database.executeCommand`

```
if (command.name === 'find' && !!command.collection && !command.sendToShell) {
	await editorManager.showDocument(context, new MongoFindResultEditor(database, command), 'cosmos-result.json', { showInNextColumn: true });
} else {
	const result = await database.executeCommand(command, context);
	if (command.name === 'findOne' && !!command.collection) {
		await editorManager.showDocument(context, new MongoFindOneResultEditor(database, command.collection, result), 'cosmos-result.json', { showInNextColumn: true });
	} else { // failover also includes chained commands
		await vscodeUtil.showNewFile(result, 'result', '.json', activeEditor.viewColumn + 1);
		await refreshTreeAfterCommand(database, command, context);
	}
}
```

I just think there's a lot of duplicate checks here and I'm concerned we're missing a case of `command.sendToShell`. If you move it all into `database.executeCommand` and integrate it correctly, I think you'll only ever have to check `command.sendToShell` once